packages/core/src/conversation/ConversationService/ConversationService.ts

```diff
@@ -562,19 +633,56 @@ export class ConversationService extends TypedEventEmitter<Events> {
     try {
       await this.mlsService.joinByExternalCommit(() => this.apiClient.api.conversation.getGroupInfo(conversationId));
     } catch (error) {
+      if (!shouldRetry) {
+        this.logger.warn("Tried to join MLS conversation but it's still broken after reset", error);
+        throw error;
+      }
+
       this.logger.warn(`Failed to join MLS conversation ${conversationId.id} via external commit`, error);
-      if (MLSService.isBrokenMLSConversationError(error)) {
+
+      if (isBrokenMLSConversationError(error)) {
         this.logger.info('Resetting MLS conversation due to broken mls conversation error', error);
-        if (!shouldRetry) {
-          this.logger.warn("Tried to join MLS conversation but it's still broken after reset", error);
-          throw error;
-        }
         return this.handleBrokenMLSConversation(conversationId);
       }
       throw error;
     }
   }
 
+  private reactToKeyMaterialUpdateFailure = async ({error, groupId}: {error: unknown; groupId: string}) => {
+    this.logger.info(`Reacting to key material update failure for group ${groupId}`);
+
+    const conversations = await this.apiClient.api.conversation.getConversationList();
+    const conversation = (conversations.found || []).find(conversation => conversation.group_id === groupId);
+
+    if (!conversation) {
+      this.logger.warn(`No conversation found for group ${groupId}`);
+      throw error;
```

Comment:
for event handlers best practice is to handle errors by caller so that it doesn't crash/propogate unexpectadly. after logging we can do an early return and handle error in the caller already suggested in previous comment.